### PR TITLE
Adding Subject support to the AuthnRequest

### PR DESCRIPTION
**Pull request by DRvanR**
As Requested for the ability to implement Generic SAML StepUp Providers as second factor for the StepUp Project (https://www.pivotaltracker.com/n/projects/1163646)

### CODE REVIEW

**Comment by DRvanR** [comment]
@rjkip poke!

File: src/SAML2/AuthnRequest.php [hunk 1]
```diff
@@ -544,4 +730,31 @@ public function toUnsignedXML()
...
+            return;
+        }
+
+        $subject = $root->ownerDocument->createElementNS(SAML2_Const::NS_SAML, 'saml:Subject');
+        $root->appendChild($subject);
+
+        if ($this->encryptedNameId === NULL) {
+            SAML2_Utils::addNameId($subject, $this->nameId);
+        } else {
+            $eid = $subject->ownerDocument->createElementNS(SAML2_Const::NS_SAML, 'saml:' . 'EncryptedID');
```
Comment (rjkip) [inline src/SAML2/AuthnRequest.php]: And why is it `'saml:' . 'EncryptedID'`?

File: tests/SAML2/AuthnRequestTest.php [hunk 2]
```diff
@@ -68,4 +71,42 @@ public function testExtensionOrdering()
...
+
+    public function testThatTheSubjectCanBeSetBySettingTheNameId()
+    {
+        $request = new SAML2_AuthnRequest();
+        $request->setNameId(array('Value' => 'user@example.org', 'Format' => SAML2_Const::NAMEID_UNSPECIFIED));
+
+        $requestAsXML = $request->toUnsignedXML()->ownerDocument->saveXML();
+        $expected = '<saml:Subject><saml:NameID Format="urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified">user@example.org</saml:NameID></saml:Subject>';
+        $this->assertContains($expected, $requestAsXML);
+    }
```
Comment (rjkip) [inline tests/SAML2/AuthnRequestTest.php]: No tests for encrypted NameIDs?

**Comment by rjkip** [comment]
Comments were addressing copy-pasted code. :+1:

**Pull request merged**